apollo-backend/src/main/java/io/logz/apollo/blockers/types/SingleRegionBlocker.java

```diff
@@ -37,13 +37,16 @@ public boolean shouldBlock(BlockerInjectableCommons blockerInjectableCommons, De
     }
 
     @Override
-    public RequestBlockerResponse shouldBlock(List<Integer> serviceIds, int numOfEnvironments) {
-        List<Integer> serviceIdsToCheck = getServicesWithinSingleRegionBlocker(serviceIds);
-        if (!serviceIdsToCheck.isEmpty()) {
-            if (numOfEnvironments > 1) {
-                return new RequestBlockerResponse(true, BLOCKER_NAME, serviceIdsToCheck);
+    public RequestBlockerResponse shouldBlock(List<Integer> serviceIds, int numOfEnvironments, String availability) {
+        if (availability.equals(singleRegionBlockerConfiguration.getAvailability())) {
```

Comment:
@matvey-mtn There is not really option of NPE because you can not create a SingleRegionBlocker without availability. I can add check for null statement on getAvailability() if you wish to.

Reply:
I see
No need in null check
all good!